internal/secretref/secret_ref.go

```diff
@@ -28,6 +30,19 @@ func ReferencesNonExistentSecret(ctx context.Context, client client.Reader, gett
 	return false
 }
 
+func ValidateAndSanitizeTLSSecret(tlsCert, tlsKey []byte) ([]byte, []byte) {
```

Comment:
Not related to secret references. Let's move this to the package where it's used.